src/metrics/mod.rs

```diff
@@ -118,3 +118,49 @@ pub fn capture_metrics(controller: &Controller) -> impl Iterator<Item = Event> {
 
     events.into_iter()
 }
+
+#[macro_export]
+/// This macro is used to emit metrics as a `counter` while simultaneously
+/// converting from absolute values to incremental values.
+///
+/// Values that do not arrive in strictly monotonically increasing order are
+/// ignored and will not be emitted.
+macro_rules! update_counter {
+    ($label:literal, $value:expr) => {{
+        use ::std::sync::atomic::{AtomicU64, Ordering};
+
+        ::lazy_static::lazy_static! {
+            static ref PREVIOUS_VALUE: AtomicU64 = AtomicU64::new(0);
```

Comment:
What's the MSRV for Vector? You could avoid `lazy_static` entirely from 1.34.0 and onward as `AtomicU64::new` is `const`.

Reply:
Since we aren't distributing vector as a library (yet?) I don't think we specify a MSRV.

Reply:
Also nice suggestion 👍